coordinator/src/main/scala/filodb.coordinator/queryplanner/ShardKeyRegexPlanner.scala

```diff
@@ -1,25 +1,37 @@
 package filodb.coordinator.queryplanner
 
 import filodb.core.metadata.Dataset
-import filodb.core.query.{ColumnFilter, QueryContext}
+import filodb.core.query.{ColumnFilter, PromQlQueryParams, QueryContext}
 import filodb.core.query.Filter.{EqualsRegex, NotEqualsRegex}
 import filodb.query.{Aggregate, BinaryJoin, LogicalPlan}
 import filodb.query.exec.{DistConcatExec, ExecPlan, InProcessPlanDispatcher, ReduceAggregateExec}
 
+/**
+ * Holder for the shard key regex matcher results.
+ *
+ * @param updatedFilters - Modified filters
+ * @param updatedQuery   - Modified query
+ */
+case class ShardKeyMatcherResult(updatedFilters: Seq[ColumnFilter], updatedQuery: String)
+
 /**
   * Responsible for query planning for queries having regex in shard column
   *
   * @param dataset         dataset
   * @param queryPlanner    multiPartition query planner
   * @param shardKeyMatcher used to get values for regex shard keys. Each inner sequence corresponds to matching regex
-  * value. For example: Seq(ColumnFilter(ws, Equals(demo)), ColumnFilter(ns, EqualsRegex(App*)) returns
-  * Seq(Seq(ColumnFilter(ws, Equals(demo)), ColumnFilter(ns, Equals(App1))), Seq(ColumnFilter(ws, Equals(demo)),
-  * ColumnFilter(ns, Equals(App2))
+  * value. For example: Seq(ColumnFilter(ws, Equals(demo)), ColumnFilter(ns, EqualsRegex(App*)) with following query
+  * metric1{ws="demo", ns=~"App*} returns,
```

Comment:
`ns=~"App.*"` in multiple places . The `.` is needed.

Reply:
Yes, somehow it got slipped through previous review. Fixing it in my PR now.